admin/resources/js/components/classification/UpdateClassification.tsx

```diff
@@ -0,0 +1,153 @@
+import { pick, upperCase } from "lodash";
+import * as React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import {
+  Classification,
+  UpdateClassificationInput,
+  useGetClassificationQuery,
+  useUpdateClassificationMutation,
+} from "../../api/generated";
+import errorMessages from "../form/errorMessages";
+import Input from "../form/Input";
+import Submit from "../form/Submit";
+
+type FormValues = UpdateClassificationInput;
+interface UpdateClassificationFormProps {
+  initialClassification: Classification;
+  handleUpdateClassification: (
+    id: string,
+    data: FormValues,
+  ) => Promise<FormValues>;
+}
+
+export const UpdateClassificationForm: React.FunctionComponent<UpdateClassificationFormProps> =
+  ({ initialClassification, handleUpdateClassification }) => {
+    const methods = useForm<FormValues>({
+      defaultValues: initialClassification,
+    });
+    const { handleSubmit, watch, reset } = methods;
+    const watchMinSalary = watch("minSalary");
+
+    const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+      const classification: FormValues = {
+        name: {
+          en: data.name?.en,
+          fr: data.name?.fr,
+        },
+        group: upperCase(data.group || ""),
+        minSalary: Number(data.minSalary),
+        maxSalary: Number(data.maxSalary),
+      };
+      return handleUpdateClassification(
+        initialClassification.id,
+        classification,
+      )
+        .then(reset) // Reset form with returned data. This resets isDirty flag.
+        .catch(() => {
+          // Something went wrong with handleUpdateClassification.
+          // Do nothing.
+        });
+    };
+    return (
+      <section>
+        <h2>Update Classification</h2>
+        <FormProvider {...methods}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <Input
+              id="name_en"
+              name="name.en"
+              label="Name: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="name_fr"
+              name="name.fr"
+              label="Name FR: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="group"
+              name="group"
+              label="Group: "
+              type="text"
+              rules={{ required: errorMessages.required }}
+            />
+            <Input
+              id="level"
+              name="level"
+              label="Level: "
+              type="number"
```

Comment:
This allows for passing negative values or passing a long value that would return an error (for example: GraphQL large integer error: Int cannot represent non 32-bit signed integer value). I've never seen a level for any classification above 8, and so here are two possible solutions with that in mind:

Add a rule where the value must be between 1 and 10
Change the field from an input to a select and only allow options for 1 to 10